.github/workflows/commit_shelter.yml

```diff
@@ -1,55 +1,146 @@
+        yum makecache;
+        yum install --nogpgcheck -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-devel libsgx-uae-service;
+        rm -f sgx_rpm_local_repo.tgz;
+        mv /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Install alinux dependency
+      if: ${{ contains(matrix.tag, 'alinux') }}
+      run: docker exec $inclavare_test bash -c 'cd /root/inclavare-containers/${{ matrix.tag }};
+        yum -y install yum-utils wget tar procps alinux-release-experimentals  protobuf-c-devel;
+        wget -c https://mirrors.openanolis.org/inclavare-containers/alinux2/sgx_rpm_local_repo.tar.gz;
+        tar xzf sgx_rpm_local_repo.tar.gz;
+        yum-config-manager --add-repo file:/root/inclavare-containers/${{ matrix.tag }}/sgx_rpm_local_repo;
+        yum makecache;
+        yum install --nogpgcheck -y libsgx-dcap-quote-verify libsgx-dcap-default-qpl libsgx-dcap-ql-devel libsgx-uae-service;
+        rm -f sgx_rpm_local_repo.tar.gz;
+        mv /root/inclavare-containers/${{ matrix.tag }}/enclave-tls /opt'
+
+    - name: Run enclave tls server for null quote
+      if: ${{ contains(matrix.sgx, 'SGX1') }}
+      run: docker exec $inclavare_test bash -c 'cd /opt/enclave-tls/bin/ && ./enclave-tls-server -a nullquote' &
+
+    - name: Run shelter for null quote
+      if: ${{ contains(matrix.sgx, 'SGX1') }}
+      run: docker exec $inclavare_test bash -c 'ls /root/inclavare-containers/${{ matrix.tag }} && cd /root/inclavare-containers/${{ matrix.tag }}/shelter && ./shelter remoteattestation  --verifier nullquote'
+        docker restart $inclavare_test;
+
+    - name: Run enclave tls server for mutual sgx_ecdsa
+      if: ${{ contains(matrix.sgx, 'SGX2') }}
+      run: |
+        docker cp /etc/sgx_default_qcnl.conf  $inclavare_test:/etc/;
+        docker exec $inclavare_test bash -c 'cd /opt/enclave-tls/bin/ && ./enclave-tls-server -l debug -a sgx_ecdsa -i 127.0.0.1 -p 1235 -m' &
+
+    - name: Run shelter for mutual sgx_ecdsa
+      if: ${{ contains(matrix.sgx, 'SGX2') }}
+      run: docker exec $inclavare_test bash -c 'ls /root/inclavare-containers/${{ matrix.tag }} && cd /root/inclavare-containers/${{ matrix.tag }}/shelter && ./shelter remoteattestation --crypto wolfcrypt --verifier sgx_ecdsa --mutual 1 --ip 127.0.0.1 --port 1235'
+        docker restart $inclavare_test;
+
+    - name: Run enclave tls server for sgx_la
+      if: ${{ contains(matrix.sgx, 'SGX2') }}
+      run: |
+        docker cp /etc/sgx_default_qcnl.conf  $inclavare_test:/etc/;
+        docker exec $inclavare_test bash -c 'cd /opt/enclave-tls/bin/ && ./enclave-tls-server -l debug -a sgx_la -t wolfssl_sgx -c wolfcrypt' &
+
+    - name: Run enclave tls server for sgx_la
+      if: ${{ contains(matrix.sgx, 'SGX2') }}
+      run: docker exec $inclavare_test bash -c 'cd /root/inclavare-containers/${{ matrix.tag }}/shelter && ./shelter remoteattestation --crypto wolfcrypt --verifier sgx_la'
```

Comment:
So why not put shelter and tls server together?

Reply:
Separating the server and the client into two steps can help us to view log information more clearly